src/main/java/uk/gov/pay/webhooks/webhook/WebhookService.java

```diff
@@ -157,19 +158,32 @@ private boolean webhookHasSubscriptionForEvent(WebhookEntity webhook, InternalEv
                 .anyMatch(internalEventNames -> internalEventNames.contains(event.eventType()));
     }
 
-    public void deleteWebhookMessagesOlderThan() {
+    public void expireWebhookMessages() {
         int maxAgeOfMessages = webhookMessageDeletionConfig.getMaxAgeOfMessages();
         int maxNumOfMessagesToExpire = webhookMessageDeletionConfig.getMaxNumOfMessagesToExpire();
+
+        List<WebhookMessageEntity> candidateWebhookMessagesForDeletion = webhookMessageDao.getWebhookMessagesOlderThan(maxAgeOfMessages);
+        List<WebhookDeliveryQueueEntity> candidateDeliveryQueueEntitiesForDeletion = 
+                webhookDeliveryQueueDao.getWebhookDeliveryQueueEntitiesOlderThan(maxAgeOfMessages);
+        List<Long> webhookMessageEntityIdsReferencedByCandidateDeliveryQueueEntities = 
+                candidateDeliveryQueueEntitiesForDeletion.stream().map(entity -> entity.getWebhookMessageEntity().getId()).collect(Collectors.toList());
         
-        var webhookMessages = webhookMessageDao.getWebhookMessagesOlderThan(maxAgeOfMessages);
-        int webhookMessagesDeleted = webhookMessageDao.deleteMessages(
-                webhookMessages.stream().limit(maxNumOfMessagesToExpire));
-        LOGGER.info(format("%s webhook messages were deleted.", webhookMessagesDeleted));
-
-        var deliveryQueueEntities = webhookDeliveryQueueDao.getWebhookDeliveryQueueEntitiesOlderThan(maxAgeOfMessages);
-        int deliveryQueueEntitiesDeleted = webhookDeliveryQueueDao.deleteDeliveryQueueEntries(
-                deliveryQueueEntities.stream().limit(maxNumOfMessagesToExpire));
+        var webhookDeliveryQueueEntitiesToDelete = candidateDeliveryQueueEntitiesForDeletion.stream().limit(maxNumOfMessagesToExpire).toList();
```

Comment:
Don't think we need to limit the number of messages from the `queue` table.

We can remove entries from `webhooks_message` that are older than 7 days and by `maxNumOfMessagesToExpire`. We can remove all associated `queue` entries.